onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -8372,6 +8372,52 @@ def aten_unique_consecutive(
     raise NotImplementedError()
 
 
+@torch_op("aten::unique", trace_only=True)
+def aten_unique(
+    self: TensorType,
+    sorted: bool = True,
+    return_inverse: bool = False,
+    return_counts: bool = False,
+    dim: Optional[int] = None,
+) -> tuple[TensorType, TensorType, TensorType]:
+    """unique(Tensor self, bool sorted=True, bool return_inverse=False, bool return_counts=False, int? dim=None) -> (Tensor, Tensor?, Tensor?)"""
+
+    if dim is None:
+        unique_values, inverse_indices, counts = aten_unique2(self, sorted, return_inverse, return_counts)
+    else:
+        unique_values, inverse_indices, counts = aten_unique_dim(self, dim, sorted, return_inverse, return_counts)
+    if return_inverse:
+        if return_counts:
+            result = unique_values, inverse_indices, counts
+        else:
+            result = unique_values, inverse_indices
+    elif return_counts:
+        result = unique_values, counts
+    else:
+        result = unique_values
+    return result
```

Comment:
I think we need to always return the same number of values. Consider returning None when they are not available?

Reply:
Doing so deviates from the behavior of `torch.unique` and causes this assertion in the unit tests to fail:
https://github.com/microsoft/onnxscript/blob/69ae7f421e2f1931e353949fa5e3a0fb23dbe622/tests/function_libs/torch_lib/ops_test.py#L251

Please advise on how to address this.

Reply:
Does `torch.ops.aten.unique` exhibit the same behavior? If it always returns three variables, consider creating a new OpInfo for `torch.ops.aten.unique` similar to https://github.com/microsoft/onnxscript/blob/a5ed07981f1e7c100c5abeb2340a841b9e81e878/tests/function_libs/torch_lib/extra_opinfo.py#L2105-L2114. You may remove the xfail with the custom OpInfo too because you may simply remove the xfail cases. 

You may adapt the sample function from https://github.com/pytorch/pytorch/blob/b948b1ad7a9cf61c9692506c60c295fd40e00f43/torch/testing/_internal/common_methods_invocations.py#L3346-L3372

Reply:
Thanks for the pointer to `extra_opinfo`.  It turns out `torch.ops.aten.unique` does not exist, but `torch.ops.aten._unique` does. Added `OpInfo` for it, `_unique2`, and `unique_dim` in 14d03b539095a80df070a2d570e023db2a4bee58